### PR TITLE
ComboboxControl: updated component to deburr options labels before filter

### LINKED ISSUE #25953: ComboboxControl: Searches should be deburred
**Is your feature request related to a problem? Please describe.**

`ComboboxControl` does not ignore diacritics, which makes it hard to search.

For example, with the list of countries in the [Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-comboboxcontrol--default), I can't type `aland` to match `Åland Islands`. 

**Describe the solution you'd like**

Queries should be [deburred](https://lodash.com/docs/4.17.15#deburr) by default, or the component should provide a deburr option via prop.

### PR DESCRIPTION
**Pull request by daddou-ma**
closes #25953

## Description
This PR ComboboxControl is updated to Deburr options labels before filter. as requested in issue #25953

### Before
image

### After
image

### Enhancements
- ComboboxControl: Deburr options labels before filter

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.

### CODE REVIEW

**Comment by adamsilverstein** [comment]
Looks good @daddou-ma - thanks for adding this.

File: packages/components/src/combobox-control/index.js [hunk 1]
```diff
@@ -68,7 +68,9 @@ function ComboboxControl( {
 		const containsMatch = [];
 		const match = inputValue.toLocaleLowerCase();
```
Comment (youknowriad) [inline packages/components/src/combobox-control/index.js:69]: Should we deburr this too?

File: docs/designers-developers/developers/data/data-core-annotations.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Namespace: `core/annotations`.
```
Comment (youknowriad) [inline docs/designers-developers/developers/data/data-core-annotations.md:8]: Seems  like  there's a lot of unrelated changes to markdown files, can we remove these changes from this PR?

**Comment by daddou-ma** [comment]
> Looks good @daddou-ma - thanks for adding this.

You're welcome @adamsilverstein :D

File: packages/components/src/combobox-control/index.js (hunk 1 above)
Comment (daddou-ma) [reply packages/components/src/combobox-control/index.js]: Yes, check the new commit 7e41bcb746878bc727897ac1661e0fc7ee824d87

**Review by adamsilverstein — APPROVED** [review]
Nice work!

**Comment by daddou-ma** [comment]
> Nice work!

Thank you @adamsilverstein ^^
